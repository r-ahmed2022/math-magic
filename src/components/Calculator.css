.cal-component {
  width: 50%;
  height: auto;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-auto-rows: minmax(100px, 1fr);
  grid-column-gap: 2px;
  grid-row-gap: 2px;
  margin: 30px 10px;
  padding: 0;
  border: none;
  box-shadow: -2px -2px 10px 5px #110;
  box-shadow: 2px 2px 10px 5px #110;
  border-radius: 5px;
}

.btn-div {
  grid-column: 1 / -1;
  display: flex;
  margin: 0;
  padding: 0;
}

input[type="text"] {
  width: 100%;
  height: 100%;
  text-align: right;
  font-size: 1.5rem;
  background: #ccc;
  color: #fff;
  border: none;
  padding: 0 10px 0 0;
  margin-left: 0;
}

.numinput::placeholder {
  color: #fff;
  font-weight: 700;
  font-size: 1.5rem;
}

.btn {
  margin: 1px;
  flex: 1 1 0;
  font-size: 1rem;
  border-style: none;
  font-weight: 700;
  border-radius: 5px;
}

.btn:hover {
  background: #ccc;
}

.txt-div {
  grid-column: 1 / -1;
  display: flex;
  margin: 0;
  justify-content: space-around;
  color: #fff;
}

.symbol {
  background: #f5913e;
}

.symbol:hover {
  background: #ff918f;
}

.btn-div-last {
  width: 100%;
  height: auto;
  grid-column: 1 / -1;
  display: grid;
  grid-template-columns: repeat(4, minmax(30px, 1fr));
  grid-auto-rows: minmax(auto, 1fr);
  grid-gap: 1px 1px;
}

.div-0 {
  grid-column: 1 / 3;
  display: flex;
  width: auto;
}

.div-last {
  grid-column: 3 / -1;
  display: flex;
  width: auto;
}

.btn-0 {
  width: 100%;
}

.btn-last {
  grid-column: 3 / 4;
}

.btn-equal {
  grid-column: 4 / -1;
}
