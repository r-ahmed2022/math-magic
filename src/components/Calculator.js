import React, { Component } from 'react';
import './Calculator.css';

export default class Calculator extends Component {
  render() {
    const {
      onclick, total, next, onchange,
    } = this.props;

    return (
      <main className="calculator-section">
        <div className="left">
          <p>
            Let&#39;s do some math....!
          </p>
        </div>
        <div className="cal-component">
          <div className="txt-div">
            <input type="text" name="numinput" id="numinput" className="numinput" maxLength="20" value={next} placeholder={total} onChange={onchange} />
          </div>
          <div className="btn-div">
            <input type="button" className="btn" value="AC" />
            <input type="button" className="btn" value="+/-" />
            <input type="button" className="btn" value="%" />
            <input type="button" className="btn symbol" value="&divide;" />
          </div>
          <div className="btn-div">
            <input type="button" className="btn" value="7" />
            <input type="button" className="btn" value="8" />
            <input type="button" className="btn" value="9" />
            <input type="button" className="btn symbol" value="x" onClick={onclick} />
          </div>
          <div className="btn-div">
            <input type="button" className="btn" value="4" />
            <input type="button" className="btn" value="5" />
            <input type="button" className="btn" value="6" />
            <input type="button" className="btn symbol" value="-" />
          </div>

          <div className="btn-div">
            <input type="button" className="btn" value="1" />
            <input type="button" className="btn" value="2" />
            <input type="button" className="btn" value="3" />
            <input type="button" className="btn symbol" value="+" />
          </div>
          <div className="btn-div-last">
            <div className="div-0">
              <input type="button" className="btn" value="0" />
            </div>
            <div className="div-last">
              <input type="button" className="btn btn-last" value="." />
              <input type="button" className="btn symbol btn-equal" value="=" />
            </div>
          </div>
        </div>
      </main>
    );
  }
}
