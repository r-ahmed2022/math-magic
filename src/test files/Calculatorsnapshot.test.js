import renderer from 'react-test-renderer';
import React from 'react';
import Calculator from '../components/Calculator';

it('Calculator renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
