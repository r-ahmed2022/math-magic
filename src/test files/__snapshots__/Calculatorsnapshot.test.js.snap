// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<main
  className="calculator-section"
>
  <div
    className="left"
  >
    <p>
      Let's do some math....!
    </p>
  </div>
  <div
    className="cal-component"
  >
    <div
      className="txt-div"
    >
      <input
        className="numinput"
        id="numinput"
        maxLength="20"
        name="numinput"
        type="text"
      />
    </div>
    <div
      className="btn-div"
    >
      <input
        className="btn"
        type="button"
        value="AC"
      />
      <input
        className="btn"
        type="button"
        value="+/-"
      />
      <input
        className="btn"
        type="button"
        value="%"
      />
      <input
        className="btn symbol"
        type="button"
        value="รท"
      />
    </div>
    <div
      className="btn-div"
    >
      <input
        className="btn"
        type="button"
        value="7"
      />
      <input
        className="btn"
        type="button"
        value="8"
      />
      <input
        className="btn"
        type="button"
        value="9"
      />
      <input
        className="btn symbol"
        type="button"
        value="x"
      />
    </div>
    <div
      className="btn-div"
    >
      <input
        className="btn"
        type="button"
        value="4"
      />
      <input
        className="btn"
        type="button"
        value="5"
      />
      <input
        className="btn"
        type="button"
        value="6"
      />
      <input
        className="btn symbol"
        type="button"
        value="-"
      />
    </div>
    <div
      className="btn-div"
    >
      <input
        className="btn"
        type="button"
        value="1"
      />
      <input
        className="btn"
        type="button"
        value="2"
      />
      <input
        className="btn"
        type="button"
        value="3"
      />
      <input
        className="btn symbol"
        type="button"
        value="+"
      />
    </div>
    <div
      className="btn-div-last"
    >
      <div
        className="div-0"
      >
        <input
          className="btn"
          type="button"
          value="0"
        />
      </div>
      <div
        className="div-last"
      >
        <input
          className="btn btn-last"
          type="button"
          value="."
        />
        <input
          className="btn symbol btn-equal"
          type="button"
          value="="
        />
      </div>
    </div>
  </div>
</main>
`;
